### PR TITLE
Update formatting of all code

### PR DESCRIPTION
**Pull request by javl**
As mentioned in #337 I'd like to update the codebase to use more consistent formatting and address some other quality of life updates. I feel this will make it a lot easier to expand this (already great) project.
I pulled the latest version of master just before this PR (thats why there are two commits) so you should be able to merge without any problems if you do decide to merge.

To be able to maintain some overview while making changes to this many files this PR contains ONLY the following changes:
* Remove unneeded white space in all files (like at the end of lines, or multiple newlines at the end of files)
* Reformats all python code using the Pycodestyle linter. 
* Adds a config file (setup.cfg) for the linter. 
 
 
You can automatically apply the linter using autopep8 on the commandline (`autopep8 --in-place`) from the main directory. If you're using VS Code to edit code, you can enable the Pycodestyle linter and add this to the config in .vscode/settings.json to use the same setup:
```
{
  "python.linting.pycodestyleEnabled": true,
  "python.linting.pycodestyleArgs": [
    "--max-line-length=180",
    "--ignore=W503,E741"
  ],
  "python.linting.enabled": true
}
```

The two Pycodestyle warnings/errors I opted to ignore are W503 (no arithmic operators (+-/ etc.) at the beginning of lines) and E741 (no variables named l o i (as `l` is used in some places)
 
Some changes I'd like to make in the future (but NOT in this commit):

* move to f-strings to improve readability
* remove old comments
* Add file extensions to all files

### CODE REVIEW

**Comment by javl** [comment]
As a note: just realized I didn't update the docs, but you might want to make changes before doing so anyway.

File: static/self.css [hunk 1]
```diff
@@ -6,7 +6,7 @@ button,
 input,
 select,
 textarea {
-    color: #222;
+    color:  #222;
```
Comment (Rotzbua) [inline static/self.css:9]: `*.css` should be ignored

Comment (javl) [reply static/self.css:9]: Not sure where this space came from. Autopep8 will by default only run on .py files. Think I might have messed up while merging some edits made in between my update and creating the PR. Can't really prevent having at least one error like this when you edit 100+ files ;)

**Comment by florianfesti** [comment]
OK; I tried to rebase and review the change but this is not that easy. Can you split the automatically generated changes into a separate patch so I can just run `autopep8` myself and see it creates the same changes, please? Also there are changes like updating the language files in there that - if at all - need to go into a separate patch.
Either you are using some other version/setting for `autopep8` than me or there are some manual changes. If they are they also need to go into a separate patch.

Sorry, for the late reply!

**Comment by florianfesti** [comment]
OK, I have played around with this a few times but never was really happy enough to merge it. For one the reformatting generates a lot of lines longer than 80 characters. The change (beside being very outdated now) is basically impossible to review properly. I guess I am just giving up and that for now. Closing.

**Comment by javl** [comment]
@florianfesti Personally I'd let go of the 80 char width, as I don't really think its needed on modern systems (I've been using 130 for a while now, not sure where I picked up that number). But I guess that's also a matter of preference. But yeah, with so many changes it's hard to review and make sure nothing broke :/

**Pull request closed without merging**